platform-validation/scripts/analyzeSystemStats.sh

```diff
@@ -0,0 +1,245 @@
+#! /bin/bash
+
+#This script is used to get Memory and CPU Usage for Provided Binaries and Containers. The script currently stores Peak and Average CPU and Memory Usage
+#Units - Memory -MB , CPU - % , Storage Size - MB
+
+#Get Location of script
+DIR="$(cd "$(dirname "$0")" && pwd)"
+STORAGE_PATH="$(realpath "${STORAGE_PATH:-$DIR}")"
+SECONDS_TO_RUN=${SECONDS_TO_RUN:-50}
+
+BINARIES="aziot-edged aziot-identityd aziot-certd aziot-keyd dockerd containerd"
+BINARYLOCATIONS="/usr/bin /usr/libexec"
+CONTAINERS="edgeHub edgeAgent SimulatedTemperatureSensor"
```

Comment:
In reality yes, the $CONTAINERS variable can be passed as a script argument so we can choose to only monitor runtime containers. There is a specific reason to add SimulatedTempSensor in the measurement 
1. In our benchmarking, when calculating size of runtime containers (Edge Hub and Edge Agent) - there is a slightly convoluted way: I need to subtract the Size of SimulatedTempSensor Container from the Total Container Size. This is because Edge Hub and Edge Agent share layers between then and I couldn't simply add the size of those two individual containers. Hence. I added monitoring of SimulatedTempSensor in there

Reply:
What about using `docker system df -v` to get size on disk that takes shared layers into account?

```bash
~$ sudo docker system df -v --format='{{range .Images}}{{printf "%s:%s, Shared:%s, Unique:%s" .Repository .Tag .SharedSize .UniqueSize}}{{pri
ntln}}{{end}}'
mcr.microsoft.com/azureiotedge-hub:1.2, Shared:105.1MB, Unique:74.76MB
mcr.microsoft.com/azureiotedge-agent:1.2, Shared:105.1MB, Unique:75.1MB
```

Reply:
That idea can also work but this involves some more complex parsing :) . For example to get total runtime container size - I need to add shared of edgeHub/edgeAgent + Unique EdgeAgent + Unique EdgeHub. With the current method, its much easier and another advantage is that we can generalize this script to get runtime characteristics of other modules as well (eg - Obs Agent)

Reply:
@damonbarry : Based on our offline conversation, I realised you were right. Now I simply monitor edgeAgent and edgeHub, get the shared and unique image size and then sum up $shared + $unique(edgeAgent) + $unique(edgeHub). 

Let me know what you think